src/main/java/com/varchenko/MikeHashMap.java

```diff
@@ -1,7 +1,219 @@
 package com.varchenko;
 
-public class MikeHashMap {
-    private static final int SIZE = 16;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 
-    private final Entry[] table = new Entry[SIZE];
-}
+public class MikeHashMap<K, V> extends AbstractMap<K, V> implements Map<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private int capacity;
+    private int currentSize;
+    private MikeEntry<K, V> current;
+    private MikeEntry<K, V>[] bucket;
+
+    public MikeHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        bucket = new MikeEntry[capacity];
+    }
+
+    public MikeHashMap(int capacity) {
+        this.capacity = capacity;
+    }
+
+
+    private int indexFor(Object k) {
+        if (k != null) {
+            int index = k.hashCode() % DEFAULT_CAPACITY;
+            return index >= 0 ? index : -index;
+        }
+        return 0;
+    }
+    private void put_not_null (K key, V value){
+        MikeEntry<K, V> mikeEntry = bucket[indexFor(key)];
+        if (mikeEntry.getKey().equals(key)) {
+            mikeEntry.setValue(value);
+        } else {
+            while (mikeEntry.getNext() != null) {
+                mikeEntry = mikeEntry.getNext();
+                currentSize++;
+            }
+            mikeEntry.setNext(new MikeEntry<>(key, value));
+        }
+    }
+
+    @Override
+    public V get(Object k) {
+        MikeEntry<K, V> mikeEntry = bucket[indexFor(k)];
+        while (mikeEntry != null) {
+            if (mikeEntry.getKey().equals(k)) {
+                return mikeEntry.getValue();
+            }
+            mikeEntry = mikeEntry.getNext();
+        }
+        for (MikeEntry<K, V> entry = bucket[0]; entry != null; entry = entry.getNext()) {
+            if (entry.getKey() == null) return entry.getValue();
+        }
+        return null;
+    }
+
+    @Override
+    public V put(K key, V value) {
+        MikeEntry<K, V> mikeEntry = bucket[indexFor(key)];
+        if (mikeEntry != null) {
+            put_not_null(key, value);
+        } else {
+            bucket[indexFor(key)] = new MikeEntry<>(key, value);
+            currentSize++;
+        }
+        return bucket[indexFor(key)].getValue();
+    }
+
+    @Override
+    public V remove(Object key) {
+        MikeEntry<K, V> mikeEntry = bucket[indexFor(key)];
+        if (mikeEntry != null) {
+            if (mikeEntry.getKey().equals(key)) {
+                mikeEntry.setValue(null);
+                currentSize--;
+            } else {
+                while (mikeEntry.getNext() != null) {
+                    mikeEntry = mikeEntry.getNext();
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> m) {
```

Comment:
Pay attention to this line. you have an issue. Think about put() method and increment.